Utilities/Loader.cpp

```diff
@@ -15,7 +14,100 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
-//#include <experimental/filesystem>
+
+std::string dataVersionOfFilename(const std::string& filename) {
+    std::string ext(filename.substr(filename.rfind(".")));
+    int orgposIfExist = filename.rfind("organisms");
+    if (orgposIfExist != -1) return filename.substr(0,orgposIfExist)+"data"+filename.substr(orgposIfExist+9);
+    else return filename.substr(0,filename.rfind("."))+"_data"+ext;
+}
+
+void followPathAndCollectFiles(std::string& curPath, unsigned int depthIntoFilterPathParts, std::vector<std::string>& filterPathParts, std::vector<std::string>& collectedFiles) {
+    std::string padding(depthIntoFilterPathParts*2,' ');
+#if defined(OS_UNIX)
+    struct dirent *fileinfo;
+    struct stat statbuf;
+    struct stat lstatbuf;
+    DIR *dir;
+    dir = opendir(curPath.c_str());
+    while( (fileinfo=readdir(dir)) ) {
+        if (strcmp(fileinfo->d_name,".")==0) continue;
+        std::string filePath(curPath+"/"+fileinfo->d_name);
+        if (curPath == "./") filePath = fileinfo->d_name;
+        stat(filePath.c_str(),&statbuf);
+        lstat(filePath.c_str(),&lstatbuf);
+        const std::regex pattern(filterPathParts[depthIntoFilterPathParts]);
+        if (S_ISDIR(statbuf.st_mode)) {
+            if ( std::regex_match(fileinfo->d_name, pattern) ) {
+                std::string newPath(curPath+"/"+fileinfo->d_name);
+                if (curPath == "./") newPath = fileinfo->d_name;
+                followPathAndCollectFiles(newPath, depthIntoFilterPathParts+1, filterPathParts, collectedFiles);
+            }
+        } else if (S_ISREG(statbuf.st_mode)) {
+            if ( std::regex_match(fileinfo->d_name, pattern) ) {
+                if (depthIntoFilterPathParts == filterPathParts.size()-1) {
+                    std::string newPath(fileinfo->d_name);
+			    		 if (curPath != "./") newPath = curPath+"/"+fileinfo->d_name;
+                    collectedFiles.push_back(newPath);
+                }
+            }
+        }
+    }
+    closedir(dir);
+#elif defined(OS_WINDOWS)
+    WIN32_FIND_DATA data;
+    HANDLE hFind;
+    std::string newCurPath(curPath);
+    DWORD curPathftyp = GetFileAttributesA(curPath.c_str());
+    if (curPathftyp & FILE_ATTRIBUTE_DIRECTORY) newCurPath = curPath+"\\*";
+    if ((hFind = FindFirstFile(newCurPath.c_str(), &data)) != INVALID_HANDLE_VALUE) {
+        do {
+            if (strcmp(data.cFileName,".")==0) continue;
+            std::string filePath(curPath+"\\"+data.cFileName);
+            if (curPath == ".") filePath = data.cFileName;
+            DWORD ftyp = GetFileAttributesA(filePath.c_str());
+            if (ftyp & FILE_ATTRIBUTE_DIRECTORY) {
+                //cout <<  "\\" << endl; // found a dir
+                if ( wildstrcmp(filterPathParts[depthIntoFilterPathParts].c_str(), data.cFileName) ) {
```

Comment:
I think this function was deleted. Has this been compiled on Windows?

Reply:
I'm unsure what you are talking about. Can you use more words? Prior to refactoring it was compiled on Windows.

Reply:
The wildstrcmp function doesn't seem to exist, (a prior commit removed it). Yet there is a call to it. Is this function defined somewhere?

Reply:
Good catch! I did not see that.